pkg/controller/master-controller-manager/kubeone/controller.go

```diff
@@ -580,137 +694,20 @@ func (r *reconciler) getKubeOneSecret(ctx context.Context, ref providerconfig.Gl
-		}
-	}
-
-	envVar := []corev1.EnvVar{}
-	volumes := []corev1.Volume{}
-
-	envVar = append(
-		envVar,
-		corev1.EnvVar{
-			Name:  "PASSPHRASE",
-			Value: string(sshSecret.Data["passphrase"]),
-		},
-	)
-
-	envFrom := []corev1.EnvFromSource{}
-	envFrom = append(
-		envFrom,
-		corev1.EnvFromSource{
-			SecretRef: &corev1.SecretEnvSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: sshSecret.Name,
-				},
-			},
-		},
-	)
-
-	vm := []corev1.VolumeMount{}
-	vm = append(
-		vm,
-		corev1.VolumeMount{
-			Name:      "ssh-volume",
-			MountPath: "/root/.ssh",
-		},
-		corev1.VolumeMount{
-			Name:      "manifest-volume",
-			MountPath: "/kubeonemanifest",
-		},
-		corev1.VolumeMount{
-			Name:      "script-volume",
-			MountPath: "/scripts",
-		},
-	)
-
-	pod := &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      KubeOneKubeconfigPod,
-			Namespace: kubeOneNamespace,
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Name:    "kubeone",
-					Image:   resources.KubeOneImage,
-					Command: []string{"/bin/sh"},
-					Args: []string{
-						"-c",
-						"./scripts/script.sh",
-					},
-					EnvFrom:      envFrom,
-					Env:          envVar,
-					Resources:    corev1.ResourceRequirements{},
-					VolumeMounts: vm,
-				},
-			},
-			Volumes: append(
-				volumes,
-				corev1.Volume{
-					Name: "manifest-volume",
-					VolumeSource: corev1.VolumeSource{
-						Secret: &corev1.SecretVolumeSource{
-							SecretName: manifestSecret.Name,
-						},
-					},
-				},
-				corev1.Volume{
-					Name: "ssh-volume",
-					VolumeSource: corev1.VolumeSource{
-						Secret: &corev1.SecretVolumeSource{
-							DefaultMode: to.Int32Ptr(256),
-							SecretName:  sshSecret.Name,
-						},
-					},
-				},
-				corev1.Volume{
-					Name: "script-volume",
-					VolumeSource: corev1.VolumeSource{
-						ConfigMap: &corev1.ConfigMapVolumeSource{
-							LocalObjectReference: corev1.LocalObjectReference{
-								Name: "kubeone",
-							},
-							DefaultMode: to.Int32Ptr(448),
-						},
-					},
-				},
-			),
-			RestartPolicy: corev1.RestartPolicyNever,
-		},
-	}
-
-	return pod, nil
-}
-
-func generateConfigMap(name, namespace, action string) *corev1.ConfigMap {
-	var scriptToRun string
+func generateConfigMap(namespace, action string) *corev1.ConfigMap {
+	var name, scriptToRun string
 	scriptToRun = resources.KubeOneScript
-	if action == ImportAction {
-		scriptToRun += "kubeone kubeconfig -m kubeonemanifest/manifest"
-	}
 
-	if action == UpgradeControlPlaneAction {
+	switch {
+	case action == ImportAction:
+		name = KubeOneImportConfigMap
+		scriptToRun += "kubeone kubeconfig -m kubeonemanifest/manifest"
+	case action == UpgradeControlPlaneAction:
+		name = KubeOneUpgradeConfigMap
 		scriptToRun += "kubeone apply -m kubeonemanifest/manifest -y"
+	case action == MigrateControlPlaneAction:
+		name = KubeOneMigrateConfigMap
+		scriptToRun += "kubeone migrate to-containerd -m kubeonemanifest/manifest"
```

Comment:
Ultra nitpick, but for written scripts, I prefer the long form of CLI flags (`--manifest` instead of `-m`)